src/main/java/org/zalando/nakadi/controller/EventTypeController.java

```diff
@@ -92,13 +76,13 @@ public EventTypeController(final EventTypeService eventTypeService,
                                     final Errors errors,
                                     final NativeWebRequest request)
             throws TopicCreationException, InternalNakadiException, NoSuchPartitionStrategyException,
-            DuplicatedEventTypeNameException, InvalidEventTypeException {
+            DuplicatedEventTypeNameException, InvalidEventTypeException, ValidationException {
         if (featureToggleService.isFeatureEnabled(DISABLE_EVENT_TYPE_CREATION)) {
             return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
         }
 
         if (errors.hasErrors()) {
-            return Responses.create(new ValidationProblem(errors), request);
+            throw new ValidationException(errors);
```

Comment:
Well, actually I was trying to avoid creating `Problem` in the controllers, and delegate that to ControllerAdvice classes instead. I agree with the first comment however, that the strategy should be consistent.

Reply:
it is not necessarily to create `Problem` here, you can create ResponseEntity
?